src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,169 @@
+
+    .. warning::
+
+        This is currently an experimental feature and API changes are to be expected.
+
+    Arguments:
+        optimization: The format in which the model should be tested while being served.
+        server: The library used to evaluate the model serving. Default is FastAPI.
+        host: The host address associated the server.
+        port: The port associated the server.
+    """
+
+    def __init__(
+        self,
+        optimization: Optional[Literal["trace", "script", "onnx", "tensorrt"]] = None,
+        server: Literal["fastapi", "ml_server", "torchserve", "sagemaker"] = "fastapi",
+        host: str = "127.0.0.1",
+        port: int = 8080,
+        timeout: int = 10,
+        exit_on_failure: bool = True,
+    ):
+        super().__init__()
+        fastapi_installed = _RequirementAvailable("fastapi")
+        if not fastapi_installed:
+            raise ModuleNotFoundError(fastapi_installed.message)
+        uvicorn_installed = _RequirementAvailable("uvicorn")
+        if not uvicorn_installed:
+            raise ModuleNotFoundError(uvicorn_installed.message)
+
+        # TODO: Add support for the other options
+        if optimization is not None:
+            raise NotImplementedError(f"The optimization {optimization} is currently not supported.")
+        # TODO: Add support for testing with those server services
+        if server != "fastapi":
+            raise NotImplementedError("Only the fastapi server is currently supported.")
+
+        self.optimization = optimization
+        self.host = host
+        self.port = port
+        self.server = server
+        self.timeout = timeout
+        self.exit_on_failure = exit_on_failure
+        self.resp: Optional[requests.Response] = None
+
+    @rank_zero_only
+    def on_train_start(self, trainer: "pl.Trainer", servable_module: "pl.LightningModule") -> None:
+        if isinstance(trainer.strategy, _NOT_SUPPORTED_STRATEGIES):
+            raise Exception(
+                f"The current strategy {trainer.strategy} used "
+                "by the trainer isn't supported for sanity serving yet."
+            )
+
+        if not isinstance(servable_module, ServableModule):
+            raise TypeError(f"The provided model should be subclass of {ServableModule.__qualname__}.")
+
+        if not is_overridden("configure_payload", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_payload` method needs to be overridden.")
+        if not is_overridden("configure_serialization", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_serialization` method needs to be overridden.")
+        if not is_overridden("serve_step", servable_module, ServableModule):
+            raise NotImplementedError("The `serve_step` method needs to be overridden.")
+
+        servable_module.trainer = None
```

Comment:
Why set it to None and to Trainer right after?

Reply:
The issue is that the trainer (or at least some of it's components - likely loggers, maybe also some other stuff) cannot be pickled. This is required for passing it to a new process.

Reply:
A comment would be helpful then